### PR TITLE
fix(router): apply order fallback before deployment affinity

### PR DESCRIPTION
**Pull request by mjnaderi**
## TLDR

Problem this solves:

- Session affinity can repin explicit order fallbacks.
- Retries can select an earlier deployment order again.

How it solves it:

- Prefilter explicit target orders before affinity callbacks.
- Add regression coverage for pinned order-one sessions.

## Pre-Submission checklist

**Please complete all items before asking a LiteLLM maintainer to review your PR**

- [x] I have added meaningful tests
- [x] My PR passes all CI/CD checks (e.g., lint, format, unit tests)
- [x] My PR's scope is as isolated as possible; it only solves 1 specific problem
- [x] I have received a Greptile **Confidence Score of at least 4/5** before requesting a maintainer review (Greptile reviews automatically once the PR is opened; only comment `@greptileai` to re-request a review after pushing changes)

## Delays in PR merge?

If you're seeing a delay in PR merge, ping the LiteLLM Team on [Slack (#pr-review)](https://join.slack.com/t/litellmossslack/shared_invite/zt-3o7nkuyfr-p_kbNJj8taRfXGgQI1~YyA).

## Screenshots / Proof of Fix

This bug is in the in-process deployment-selection pipeline and requires no provider request. On base commit `abf18f8`, affinity narrows the candidates to the cached order-1 deployment before `_target_order=2` is applied. Because no order-2 candidate remains, the order helper returns the pinned deployment unchanged. On fix commit `39a58ba`, the explicit target order narrows candidates first, so the stale order-1 pin cannot match.

```console
$ pytest tests/test_litellm/test_router_order_fallback.py -q
...............                                                          [100%]
15 passed
```

## Type

🐛 Bug Fix

## Changes

`async_get_healthy_deployments` now consumes and applies an explicit `_target_order` immediately after health, cooldown, and blocked-deployment filtering. This makes the fallback order a hard retry constraint before `DeploymentAffinityCheck` runs. The existing final order filter remains in place for normal selection, preserving routing-plugin and default-order behavior.

The regression test primes session affinity to order 1, verifies an ordinary selection remains pinned there, then verifies an order-2 fallback selects the order-2 deployment.

### Final Attestation

- [x] The tests check the right things, including the edge cases, and regressions in the respective real-world customer use-cases are not possible after this PR

### CODE REVIEW

File: litellm/router.py [hunk 1]
```diff
@@ -10619,7 +10627,6 @@ async def async_get_healthy_deployments(
         )
 
         ## ORDER FILTERING ## -> if user set 'order' in deployments, return deployments with lowest order (e.g. order=1 > order=2)
-        _target_order = (request_kwargs or {}).pop("_target_order", None)
         healthy_deployments = litellm.utils._get_order_filtered_deployments(
             cast(List[Dict], healthy_deployments), target_order=_target_order
         )
```
[the remark this answers is not in the document]
Comment (mjnaderi) [reply litellm/router.py]: Fixed in acc6f7f: the late call is now guarded by `_target_order is None`, and its comment clarifies that it only performs default lowest-order selection. The explicit fallback target is filtered once, before affinity.

**Comment by mjnaderi** [comment]
@greptileai

**Comment by mjnaderi** [comment]
@greptileai
